plugins/toolkit/jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/rds/auth/IamAuth.kt

```diff
@@ -62,14 +82,44 @@ class IamAuth : DatabaseAuthProviderCompatabilityAdapter {
                 val credentials = getCredentials(connection)
                 DatabaseCredentialsAuthProvider.applyCredentials(connection, credentials, true)
             } catch (e: Throwable) {
-                result = Result.Failed
-                throw e
+                if (e is IllegalStateException && e.message?.contains("Token refresh started before session initialized") == true) {
```

Comment:
we should not be keying off exception messages

Reply:
We can throw another exception instead of `IllegalStateException` in BearerTokenProvider. But this could cause issues with the other methods handling the exceptions.

Reply:
Fixed and pushed the commit